docker/index.js

```diff
@@ -2,100 +2,240 @@ const express = require('express');
-    const pickedOptions = (({ prompt, seed, guidance_scale }) => ({ prompt, seed, guidance_scale }))(options);
-    pickedOptions = {...pickedOptions, ...{"num_images_per_prompt": 1}};
-    
-    // pickedOptions = {
-    //     prompt: ...,
-    //     seed: ...,
-    //     guidance_scale: ...,
-    //     num_images_per_prompt: 1
-    // }
-    
-    const SDResult = await axios.get(`${SD_INPAINTING_ENDPOINT}/tasks/${task_id}`, pickedOptions);
-    console.log(JSON.stringify(SDResult.data,null,2));
-
-    if (SDResult.data.status !== "completed") {
-        console.log(`Task ${task_id} is not completed!`);
-        res.send(`Task ${task_id} is not completed!`);
-        return;
-    }
-    
-
-    //pre-check the output path
-    const outputPath = formatPath([...parsedInputPath.slice(0, parsedInputPath.length - 1), "signed_data"]);
-    const result = await ain.db.ref(outputPath).setValue({
-      value: `${JSON.stringify(SDResult.data, null, 2)}`,
-      nonce: -1,
-    })
-    .catch((e) => {
-      console.error(`setValue failure:`, e);
-    });
-  
-});
+	const { transaction } = req.body;
+	const value = transaction.tx_body.operation.value;
+	const taskId = value.params.task_id;
+	if(cache.get(taskId)){
+		cache.ttl(taskId, 60);
+		return;
+	}
+
+	// if request is first request, set cache 
+	cache.set(taskId, true, 60);
+
+	// for catch error
+	let uploadMetadataRes; 
+	let uploadImgRes;
+
+	const inputPath = transaction.tx_body.operation.ref;
+	const parsedInputPath = parsePath(inputPath);
+
+	// pre-check the output path
+	const outputPath = await formatPath([...parsedInputPath.slice(0, parsedInputPath.length - 1), "verify"]);
+	const errorPath = formatPath([...parsedInputPath.slice(0, parsedInputPath.length - 1), "error"]);
+
+	// init pinata sdk
+	const pinata = new pinataSDK({ pinataApiKey, pinataSecretApiKey });
+
+	// get image file from url
+	const imageDataResponse = await axios.get(value.params.tempImageUrl, {
+		responseType: "arraybuffer",
+	});
+
+	// for test, if develop front-end, using imageDataResponse
+	const imageBuffer = fs.readFileSync(__dirname + "/test.png");
+
+	// image file to readable stream
+	const imageDataStream = new Readable();
+	imageDataStream.push(imageBuffer);
+	imageDataStream.push(null);
+
+	// upload image to pinata
+	const options = {
+		pinataMetadata: {
+			name: `${taskId}_image`,
+		},
+		pinataOptions: {
+			cidVersion: 0,
+		},
+	};
+
+	// upload image to ipfs
+	try{
+		uploadImgRes = await pinata.pinFileToIPFS(imageDataStream, options)
+	}
+	catch(e) {
+		console.error(e);
+		ain.db.ref(errorPath).setValue({
+			value: {
+				state:"Error",
+				msg:"Image upload fail. check your inforamtion of Image"
+			},
+		})
+		.then(res => console.log(res))
+		.catch((e) => {
+			console.error(`setValue failure:`, e);
+			res.status(502).send("image upload fail");
+		});
+
+		return;
+	}
+
+	// metadata will be writed in ipfs
+	const metadata = {
+		attributes: value.params.old_attributes,
```

Comment:
We need to parse `old_attribute` in original format(contains list) when uploading IPFS_metadataV2.

Reply:
oh it was fixed :)